### PR TITLE
31 unit testing

### LINKED ISSUE #31: Unit Testing
- pytest test scripts for data processing, model training, and evaluation.
- Documentation of the testing process and examples of test cases.

### PR DESCRIPTION
**Pull request by allanzguan**
31 unit testing

### CODE REVIEW

File: tests/test_data.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import pytest
+import torch
+from torchvision import transforms
+from torchvision.datasets import MNIST
```
Comment (HenrydeB) [inline tests/test_data.py]: We did not use MNIST for our data, we have our own data

File: tests/test_data.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+dataset_path = _PATH_DATA+"/processed"
+
+
+N_train = 60000
+N_test = 10000
+
+mist_transform = transforms.Compose([transforms.ToTensor()])
+
+@pytest.fixture
+def train_dataset():
```
Comment (HenrydeB) [inline tests/test_data.py]: These need to reflect how we train our model in our project. The code in xgboost

**Review by HenrydeB — CHANGES_REQUESTED** [review]
I'm not super familiar with Python unit testing outside of this class, but it look like the main thing you need to update is the `train_data.py` the other stuff looks fine to me, but I'd like @msoria17 to take a look too

File: tests/test_data.py (hunk 1 above)
Comment (allanzguan) [reply tests/test_data.py]: Those are removed

File: tests/test_data.py (hunk 2 above)
Comment (allanzguan) [reply tests/test_data.py]: Fixed

**Review by HenrydeB — APPROVED** [review]

**Pull request merged**